WS2012R2/lisa/remote-scripts/ica/perf_ntttcp_client.sh

```diff
@@ -510,28 +510,135 @@ if [ -d  $HOME/$log_folder ]; then
+         "ulimit -n 20480 && ntttcp -u -b ${NTTTCP_BUFFER} -r ${SERVER_IP} -P ${num_threads_P}  -e ${ipVersion} > ${ntttcp_recv_log}"
+        ssh -i "$HOME/.ssh/${SSH_PRIVATE_KEY}" -f -o StrictHostKeyChecking=no ${SERVER_OS_USERNAME}@${STATIC_IP2} \
+         "lagscope -r${SERVER_IP} ${ipVersion}"
+         tcp_con_logs="$HOME/$log_folder/tcp-connections-p${current_test_threads}.log"
+        ssh -i "$HOME/.ssh/${SSH_PRIVATE_KEY}" -f -o StrictHostKeyChecking=no ${SERVER_OS_USERNAME}@${STATIC_IP2} \
+         "for ((i=1;i<=$TEST_DURATION;i++)); do ss -ta | grep ESTA | grep -v ssh | wc -l >> $tcp_con_logs;done"
+        ssh -i "$HOME/.ssh/${SSH_PRIVATE_KEY}" -f -o StrictHostKeyChecking=no ${SERVER_OS_USERNAME}@${STATIC_IP2} \
+         "sar -n DEV 1 ${TEST_DURATION} > $HOME/$log_folder/sar-receiver-p${current_test_threads}.log"
+        ssh -i "$HOME/.ssh/${SSH_PRIVATE_KEY}" -f -o StrictHostKeyChecking=no ${SERVER_OS_USERNAME}@${STATIC_IP2} \
+         "dstat -dam 1 ${TEST_DURATION} > $HOME/$log_folder/dstat-receiver-p${current_test_threads}.log"
+        ssh -i "$HOME/.ssh/${SSH_PRIVATE_KEY}" -f -o StrictHostKeyChecking=no ${SERVER_OS_USERNAME}@${STATIC_IP2} \
+         "mpstat -P ALL 1 ${TEST_DURATION} > $HOME/$log_folder/mpstat-receiver-p${current_test_threads}.log"
+
+        sar -n DEV 1 "${TEST_DURATION}" > "$HOME/$log_folder/sar-sender-p${current_test_threads}.log" &
+        dstat -dam 1 "${TEST_DURATION}" > "$HOME/$log_folder/dstat-sender-p${current_test_threads}.log" &
+        mpstat -P ALL 1 "${TEST_DURATION}" > "$HOME/$log_folder/mpstat-sender-p${current_test_threads}.log" &
+
+        lagscope -s "${SERVER_IP}" -t "${TEST_DURATION}" -V  "${ipVersion}" > "./$log_folder/lagscope-ntttcp-p${num_threads_P}X${num_threads_n}.log" &
+        ntttcp -u -b "${NTTTCP_BUFFER}" -s "${SERVER_IP}" -P "$num_threads_P" -n "$num_threads_n" -t "${TEST_DURATION}" "${ipVersion}" -f${port} > \
+         "./$log_folder/ntttcp-sender-p${num_threads_P}X${num_threads_n}.log"
+        sts=$?
+
+        current_tx_bytes=$(get_tx_bytes "$ETH_NAME")
+        current_tx_pkts=$(get_tx_pkts "$ETH_NAME")
+        bytes_new=$((current_tx_bytes - previous_tx_bytes))
+        pkts_new=$((current_tx_pkts - previous_tx_pkts))
+        avg_pkt_size=$(echo "scale=2;$bytes_new/$pkts_new/1024" | bc)
+        Throughput=$(echo "scale=2;$bytes_new/$TEST_DURATION*8/1024/1024/1024" | bc)
+        previous_tx_bytes=$current_tx_bytes
+        previous_tx_pkts=$current_tx_pkts
+
+        echo "Throughput (gbps): $Throughput"
+        echo "average packet size: $avg_pkt_size"
+        printf "%4s  %8.2f  %8.2f\\n" "${current_test_threads}" "${Throughput}" "${avg_pkt_size}" >> "${eth_log}"
+
+        echo "current test finished. wait 10 seconds for next one... "
+        sleep 10
+        pkill -x ntttcp
+        pkill -x lagscope
+        pkill -f ESTA
+        pkill -x sar
+        pkill -x dstat
+        pkill -x mpstat
```

Comment:
the pkill thing is a very dangerous way of killing process, without any checks or traceability